### PR TITLE
[Bugfix][GLM-5.3-Flash] Fix fp8_fp4_paged_mqa_logits MQA kpool varlen path for speculative decode

### PR DESCRIPTION
**Pull request by wongsingfo**
## Purpose

Pass decode_metadata.indices to fp8_fp4_paged_mqa_logits in the kpool sparse attention indexer decode path.

For flattened speculative decode, the indexer metadata builder flattens multiple query tokens from the same request into individual rows and keeps request indices so that those rows can still be associated with their original request. See DeepseekV32IndexerMetadataBuilder in vllm/v1/attention/backends/mla/indexer.py.

On SM100, DeepGEMM supports varlen paged MQA logits. It uses consecutive equal request indices to identify query rows belonging to the same request and process them together.

However, the kpool logits call did not pass decode_metadata.indices to fp8_fp4_paged_mqa_logits. As a result, the metadata could be generated for the varlen path (is_varlen=true), while the consuming DeepGEMM kernel was instantiated as is_varlen=false with next_n=1.

This mismatch does not affect correctness, but significantly hurts performance because each flattened query row is effectively handled independently instead of using the varlen grouping information.

This issue is specific to SM100. The varlen paged MQA logits path is currently enabled only for SM100; other architectures such as SM90 do not use this DeepGEMM varlen path.

## Test Plan

Run an end-to-end test with the GLM-5.3-Flash model using the following configuration:

Context length: 120K
Batch size: 40
Speculative decoding steps: 5

Use Nsight Systems (nsys) to capture the execution trace and compare the sm100_paged_mqa_logits kernel latency before and after the change.

## Test Result

The change enables the varlen kernel (is_varlen=true) and reduces the kernel latency from 214.304 μs to 51.808 μs, corresponding to a 4.14× speedup.

```
before: 
void deep_gemm::sm100_paged_mqa_logits<(bool)0, (unsigned int)1, (unsigned int)32, (unsigned int)128, (unsigned int)64, (bool)1, (bool)0, (unsigned int)3, (unsigned int)5, (unsigned int)256, (unsigned int)16, (unsigned int)128, (unsigned int)256, float, float, (unsigned int)2>(unsigned int, unsigned int, unsigned int, const unsigned int *, T14 *, const unsigned int *, const unsigned int *, const unsigned int *, CUtensorMap_st, CUtensorMap_st, CUtensorMap_st, CUtensorMap_st, CUtensorMap_st)
Begins: 1.77108s
Ends: 1.7713s (+214.304 μs)
grid:  <<<152, 1, 1>>>
block: <<<384, 1, 1>>>

after:
void deep_gemm::sm100_paged_mqa_logits<(bool)0, (unsigned int)1, (unsigned int)32, (unsigned int)128, (unsigned int)64, (bool)1, (bool)1, (unsigned int)3, (unsigned int)5, (unsigned int)256, (unsigned int)16, (unsigned int)128, (unsigned int)256, float, float, (unsigned int)2>(unsigned int, unsigned int, unsigned int, const unsigned int *, T14 *, const unsigned int *, const unsigned int *, const unsigned int *, CUtensorMap_st, CUtensorMap_st, CUtensorMap_st, CUtensorMap_st, CUtensorMap_st)
Begins: 1.505s
Ends: 1.50505s (+51.808 μs)
grid:  <<<152, 1, 1>>>
block: <<<384, 1, 1>>>
```

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Comment by wongsingfo** [comment]
Closing this as already covered in #55270.
